geoips/plugins/modules/output_checkers/netcdf.py

```diff
@@ -241,6 +220,64 @@ def outputs_match(plugin, output_product, compare_product):
     return True
 
 
+def log_object_diff_values(object1, object2, compare_key):
+    r"""
+    Log differences two objects via key and returns messages detailing differences.
+
+    Computes the maximum, minimum, and mean differences for the values associated
+    with a specified key in two objects. Generates messages for any non-zero
+    differences and logs these messages. The messages are also returned as a list
+    of strings for further use.
+
+    Parameters
+    ----------
+    object1 : object
+        The first object containing the key for comparison. Must support
+        subscripting with the compare key and arithmetic operations.
+    object2 : object
+        The second object for comparison. Must also support subscripting with the
+        compare key and arithmetic operations.
+    compare_key : str
+        The key for the values to be compared between `object1` and `object2`.
+
+    Returns
+    -------
+    messages : list of str
+        A list of messages indicating non-zero min, max, and mean differences for
+        the values associated with `compare_key`, formatted as strings.
+
+    Notes
+    -----
+    Only logs and returns messages for non-zero differences. If there are no non-zero
+    differences, an empty list is returned.
+
+    Uses `log_with_emphasis` to log the messages.
+
+    Examples
+    --------
+    Given two objects `obj1` and `obj2` with a key `temp`:
+
+    >>> obj1 = {'temp': np.array([1, 2, 3])}
+    >>> obj2 = {'temp': np.array([2, 3, 4])}
+    >>> log_object_diff_values(obj1, obj2, 'temp')
+    ['mindiff temp: -1\\n', 'maxdiff temp: -1\\n', 'meandiff temp: -1.0\\n']
+
+    This will log and return messages about differences in `temp` values.
+    """
+    maxdiff = (object2[compare_key] - object1[compare_key]).max()
+    mindiff = (object2[compare_key] - object1[compare_key]).min()
+    meandiff = (object2[compare_key] - object1[compare_key]).mean()
+    messages = []
+    if mindiff != 0:
+        messages.append(f"mindiff {compare_key}: {mindiff}\n")
+    if maxdiff != 0:
+        messages.append(f"maxdiff {compare_key}: {maxdiff}\n")
+    if meandiff != 0:
```

Comment:
I think that, given when this function will be called, we can remove these if statements. The only thing they end up doing is cause the function to error if we call it with identical objects for `object1` and `object2`. When removed, the function works as expected, even when the objects are identical.